File: Data Structures & Algorithms/reverse-string/submission-0.java
class Solution {
    public void reverseString(char[] s) {
        int p0 = 0;
        int p1 = s.length - 1;

        while (p0 < p1) {
            char c = s[p0];
            s[p0] = s[p1];
            s[p1] = c;
            p0++;
            p1--;
        }
    }
}

File: Data Structures & Algorithms/multiply-strings/submission-0.py
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        digits = {
            "1": 1,
            "2": 2,
            "3": 3,
            "4": 4,
            "5": 5,
            "6": 6,
            "7": 7,
            "8": 8,
            "9": 9,
            "0": 0,
        }

        def toNum(n):
            res = 0
            for i in range(len(n) - 1, -1, -1):
                res += pow(10, (len(n) - i - 1)) * digits[n[i]]
            return res
        
        n1 = toNum(num1)
        n2 = toNum(num2)
        
        return str(n1 * n2)